src/types.ts

```diff
@@ -52,28 +59,44 @@ export interface ISession {
   state: string;
 }
 
-export type Emailer = string;
-
 export type EmailHeader = string;
 
 export type Attachment = File;
 
-export interface IEmail {
+export interface IEmailProperties {
   id: string;
   blobId: string;
   threadId: string;
   mailboxIds: { [key: string]: boolean };
-  keywords: { [key: string]: boolean };
-  from: Emailer[] | null;
-  to: Emailer[] | null;
+  keywords: IEmailKeywords;
+  from: IEmailAddress[] | null;
+  to: IEmailAddress[] | null;
   subject: string;
-  date: Date;
   size: number;
   preview: string;
   attachments: Attachment[] | null;
   createdModSeq: number;
   updatedModSeq: number;
-  deleted: Date | null;
```

Comment:
What are these createdModSeq updatedModSeq properties?

Looking at it in the spec I have 0 matches...

Is it necessary?

(I agree, this is not related to your work)

Reply:
I was wondering the same, but decided to lessen the overhead of changes. I was unable to locate those as well. However, they are present here https://jmap.io/server.html#3-mailboxes and as far as I understand, it is (draft aka not based on confirmed standard). Maybe @alagane can add some more detail in regards to it?